infra/main.bicep

```diff
@@ -2,51 +2,138 @@ targetScope = 'subscription'
+    applicationInsightsName: monitoring.outputs.applicationInsightsName
+    daprEnabled: true
+    manegeIdentityClientId: serviceBusAccess.outputs.managedIdentityClientlId
+  }
+}
+
+module checkoutApp './app/checkout.bicep' = {
+  name: 'api-resources'
+  scope: rg
+  params: {
+    name: !empty(checkoutContainerAppName) ? checkoutContainerAppName : '${abbrs.appContainerApps}${checkoutServiceName}-${resourceToken}'
+    serviceName: checkoutServiceName
+    containerRegistryName: appEnv.outputs.registryName
+    imageName: checkoutImageName
+    location: location
+    containerAppsEnvironmentName: appEnv.outputs.environmentName
+    managedIdentityName: serviceBusAccess.outputs.managedIdentityName
   }
+  dependsOn: [
+    ordersApp  // Deploy the subscriber first and then deploy the publisher
+  ]
 }
 
-output AZURE_KEY_VAULT_ENDPOINT string = resources.outputs.AZURE_KEY_VAULT_ENDPOINT
-output SERVICEBUS_ENDPOINT string = resources.outputs.SERVICEBUS_ENDPONT
-output APPINSIGHTS_INSTRUMENTATIONKEY string = resources.outputs.APPINSIGHTS_INSTRUMENTATIONKEY
-output AZURE_CONTAINER_REGISTRY_ENDPOINT string = resources.outputs.AZURE_CONTAINER_REGISTRY_ENDPOINT
-output AZURE_CONTAINER_REGISTRY_NAME string = resources.outputs.AZURE_CONTAINER_REGISTRY_NAME
-output APP_CHECKOUT_BASE_URL string = resources.outputs.CHECKOUT_APP_URI
-output APP_ORDERS_BASE_URL string = resources.outputs.ORDERS_APP_URI
-output APP_APPINSIGHTS_INSTRUMENTATIONKEY string = resources.outputs.APPINSIGHTS_INSTRUMENTATIONKEY
+module ordersApp './app/orders.bicep' = {
+  name: 'web-resources'
+  scope: rg
+  params: {
+    name: !empty(ordersContainerAppName) ? ordersContainerAppName : '${abbrs.appContainerApps}${ordersServiceName}-${resourceToken}'
+    location: location
+    imageName: ordersImageName
+    containerRegistryName: appEnv.outputs.registryName
+    containerAppsEnvironmentName: appEnv.outputs.environmentName
+    serviceName: ordersServiceName
+    managedIdentityName: serviceBusAccess.outputs.managedIdentityName
+  }
+}
+
+
+output SERVICE_CHECKOUT_NAME string = checkoutApp.outputs.SERVICE_CHECKOUT_NAME
+output SERVICE_ORDERS_NAME string = ordersApp.outputs.SERVICE_ORDERS_NAME
+output ORDERS_APP_URI string = ordersApp.outputs.ORDERS_URI
+output SERVICEBUS_ENDPOINT string = serviceBusResources.outputs.SERVICEBUS_ENDPOINT
+output SERVICEBUS_NAME string = serviceBusResources.outputs.serviceBusName
+output APPINSIGHTS_INSTRUMENTATIONKEY string = monitoring.outputs.applicationInsightsInstrumentationKey
+output APPLICATIONINSIGHTS_NAME string = monitoring.outputs.applicationInsightsName
+output AZURE_CONTAINER_ENVIRONMENT_NAME string = appEnv.outputs.environmentName
+output AZURE_CONTAINER_REGISTRY_ENDPOINT string = appEnv.outputs.registryLoginServer
+output AZURE_CONTAINER_REGISTRY_NAME string = appEnv.outputs.registryName
+output AZURE_MANAGED_IDENTITY_NAME string = serviceBusAccess.outputs.managedIdentityName
```

Comment:
Should we create user managed identity first in main, and then pass that to all the resources that need it?  it's weird to me that it originates from Service Bus (if I understand this).

Reply:
agreed changing it.